core/src/apps/cardano/layout.py

```diff
@@ -157,6 +223,7 @@ async def confirm_transaction(
     fee: int,
     protocol_magic: int,
     ttl: int,
+    validity_interval_start: Optional[int],
```

Comment:
I forgot to update the type annotation. The issue here is that although the code has type annotations they are not being enforced in this file because Trezor is still migrating to it and it would trigger too much error, so they serve just as live documentation right now. Refactor of the memory management is a good opportunity to get that finally working